airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -102,67 +142,88 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
             return {self.cursor_field: max(latest_benchmark, current_stream_state[self.cursor_field])}
         return {self.cursor_field: latest_benchmark}
 
-    def _create_prepared_request(
-        self, path: str, headers: Mapping = None, params: Mapping = None, json: Any = None, data: Any = None
-    ) -> requests.PreparedRequest:
-        """
-        Override to prepare request for AWS API.
-        AWS signature flow require prepared request to correctly generate `authorization` header.
-        Add `auth` arg to sign all the requests with AWS signature.
-        """
 
-        return self._session.prepare_request(
-            requests.Request(method=self.http_method, url=self.url_base + path, headers=headers, params=params, auth=self._aws_signature)
+class ReportsAmazonSPStream(AmazonSPStream, ABC):
+    primary_key = "reportId"
+    path_prefix = f"/reports/{REPORTS_API_VERSION}/reports"
+    wait_seconds = 30
+
+    def should_retry(self, response: requests.Response) -> bool:
+        should_retry = super().should_retry(response)
+        if not should_retry:
+            should_retry = response.json().get(self.data_field, {}).get("processingStatus") in ["IN_QUEUE", "IN_PROGRESS"]
+        return should_retry
+
+    def backoff_time(self, response: requests.Response):
+        return self.wait_seconds
+
+    def _create_report(self, stream_state: Mapping[str, Any] = None) -> Mapping[str, Any]:
+        request_headers = self.request_headers(stream_state=stream_state)
+        report_data = {"reportType": self.name, "marketplaceIds": self.marketplace_ids}
+        create_report_request = self._create_prepared_request(
+            method="POST",
+            path=self.path_prefix,
+            headers=dict(request_headers, **self.authenticator.get_auth_header()),
+            data=json_lib.dumps(report_data),
         )
+        report_response = self._send_request(create_report_request, {})
 
-    def request_headers(self, *args, **kwargs) -> Mapping[str, Any]:
-        return {"content-type": "application/json"}
-
+        time.sleep(self.wait_seconds)
 
-class ReportsBase(AmazonSPStream, ABC):
-    primary_key = "reportId"
-    cursor_field = "createdTime"
-    replication_start_date_field = "createdSince"
-    next_page_token_field = "nextToken"
-    page_size_field = "pageSize"
+        return report_response.json()[self.data_field]
 
-    def path(self, **kwargs):
-        return f"/reports/{REPORTS_API_VERSION}/reports"
+    def path(self, stream_state: Mapping[str, Any] = None, **kwargs):
+        super().path()
+        return f"{self.path_prefix}/{self._create_report(stream_state)[self.primary_key]}"
```

Comment:
```suggestion
        report_id = self._create_report(stream_state)[self.primary_key]
        return f"{self.path_prefix}/{report_id}"
```
could you remind me why we can't do this as a slice? I see slices approach as more clean